### PR TITLE
Deprecate `ActiveModel::Errors` `add_on_empty` and `add_on_blank` methods

### PR DESCRIPTION
**Pull request by morgoth**
Follow up to https://github.com/rails/rails/pull/18634#issuecomment-74946516

### CODE REVIEW

**Comment by morgoth** [comment]
I'm not sure if we should delete this test https://github.com/rails/rails/pull/18996/files#diff-399b5cddc709ef96edf30de2328934deR56 as it doesn't have any equivalent in regular validations and `:empty` symbol is used only in deprecated `add_on_empty` method

**Comment by senny** [comment]
@morgoth let's remove that test when we remove the deprecated methods and keep it around for now.

**Comment by senny** [comment]
Even though we do not provide a direct replacement the deprecation message should give a hint where to look for similar behavior. It should be trivial to figure out what to change or where to look for more details.

**Comment by morgoth** [comment]
@senny I added tip to use `PresenceValidator`. In `add_on_empty` it's almost true ;-), as I'm not sure if we should describing implementation details here.

**Comment by senny** [comment]
The thing is that `PresenceValidator` is `# :nodoc:`. Don't think we should encourage it's use.

**Comment by morgoth** [comment]
What about "To achieve the same use validates_presence_of with :empty message"?

**Comment by senny** [comment]
If you have used `add_on_empty` its likely in a custom validation method with logic attached to it. That's probably why they didn't choose to use a `validates_presence_of`. Maybe combined with an `:if` option?

/cc @rafaelfranca

**Comment by rafaelfranca** [comment]
Yeah, it is more a kind of internal method to be used in custom validations. I think it is better to suggest people to write the actual code:

```ruby
errors.add(attribute, :empty, options) if value.nil? || value.empty?
```

**Comment by morgoth** [comment]
@rafaelfranca I changed it to your suggestion

File: activemodel/lib/active_model/errors.rb [hunk 1]
```diff
@@ -361,6 +368,12 @@ def add_on_empty(attributes, options = {})
     #   person.errors.messages
     #   # => {:name=>["can't be blank"]}
     def add_on_blank(attributes, options = {})
+      ActiveSupport::Deprecation.warn(<<-MESSAGE.squish)
+        ActiveModel::Errors#set is deprecated and will be removed in Rails 5.1
+
+        To achieve the same use validates_presence_of
```
Comment (rafaelfranca) [inline activemodel/lib/active_model/errors.rb]: Same here but using `blank?`

**Comment by morgoth** [comment]
done

**Comment by senny** [comment]
@morgoth thank you :yellow_heart:

**Pull request merged**